mysite/chat/serializers.py

```diff
@@ -0,0 +1,298 @@
+import injections
+from rest_framework import serializers
+from django.core.urlresolvers import reverse
+
+from .models import Message, Chat, ChatDivider
+from .services import ProgressHandler
+from ct.models import UnitLesson, Response
+from accounts.models import Instructor
+
+
+class InternalMessageSerializer(serializers.ModelSerializer):
+    """
+    Serializer for addMessage list representation.
+    """
+    html = serializers.CharField(source='get_html', read_only=True)
+    name = serializers.CharField(source='get_name', read_only=True)
+    avatar = serializers.SerializerMethodField()
+
+    class Meta:
+        model = Message
+        fields = (
+            'id',
+            'type',
+            'name',
+            'userMessage',
+            'avatar',
+            'html'
+        )
+
+    def get_avatar(self, obj):
+        if not obj.userMessage:
+            try:
+                return obj.chat.instructor.instructor.icon_url
+            except Instructor.DoesNotExist:
+                pass
+
+
+class InputSerializer(serializers.Serializer):
+    """
+    Serializer for input description for next message.
+    """
+    type = serializers.CharField(max_length=16, read_only=True)
```

Comment:
> I would recommend not to use the keyword type as a field name.

I would recommend the same :)
But it is requirement from frontend app.
I can rewrite JS offcourse but it would be great to know what exactly can happen in this particular case. Thanks.